pop-api/examples/fungibles/lib.rs

```diff
@@ -1,108 +1,291 @@
+			if contract == spender || value == 0 {
+				return Ok(());
+			}
+			api::decrease_allowance(self.id, spender, value).map_err(Psp22Error::from)?;
+			let value = self.allowance(contract, spender);
+			self.env().emit_event(Approval { owner: contract, spender, value });
+			Ok(())
+		}
+	}
+
+	impl Psp22Metadata for Fungible {
+		/// Returns the token name.
+		#[ink(message)]
+		fn token_name(&self) -> Option<String> {
+			api::token_name(self.id)
+				.unwrap_or_default()
+				.and_then(|v| String::from_utf8(v).ok())
 		}
 
+		/// Returns the token symbol.
 		#[ink(message)]
-		pub fn token_name(&self, token: TokenId) -> Result<Vec<u8>> {
-			api::token_name(token)
+		fn token_symbol(&self) -> Option<String> {
+			api::token_symbol(self.id)
+				.unwrap_or_default()
+				.and_then(|v| String::from_utf8(v).ok())
 		}
 
+		/// Returns the token decimals.
+		#[ink(message)]
+		fn token_decimals(&self) -> u8 {
+			api::token_decimals(self.id).unwrap_or_default()
+		}
+	}
+
+	impl Psp22Mintable for Fungible {
+		/// Creates `value` amount of tokens and assigns them to `account`, increasing the total
+		/// supply.
+		///
+		/// # Parameters
+		/// - `account` - The account to be credited with the created tokens.
+		/// - `value` - The number of tokens to mint.
 		#[ink(message)]
-		pub fn token_symbol(&self, token: TokenId) -> Result<Vec<u8>> {
-			api::token_symbol(token)
+		fn mint(&mut self, account: AccountId, value: Balance) -> Result<(), Psp22Error> {
+			self.ensure_owner()?;
+			// No-op if `value` is zero.
+			if value == 0 {
+				return Ok(());
+			}
+			api::mint(self.id, account, value).map_err(Psp22Error::from)?;
+			self.env().emit_event(Transfer { from: None, to: Some(account), value });
+			Ok(())
+		}
+	}
+
+	impl Psp22Burnable for Fungible {
+		/// Destroys `value` amount of tokens from `account`, reducing the total supply.
+		///
+		/// # Parameters
+		/// - `account` - The account from which the tokens will be destroyed.
+		/// - `value` - The number of tokens to destroy.
+		#[ink(message)]
+		fn burn(&mut self, account: AccountId, value: Balance) -> Result<(), Psp22Error> {
+			self.ensure_owner()?;
+			// No-op if `value` is zero.
+			if value == 0 {
+				return Ok(());
+			}
+			api::burn(self.id, account, value).map_err(Psp22Error::from)?;
+			self.env().emit_event(Transfer { from: Some(account), to: None, value });
+			Ok(())
+		}
+	}
+
+	impl Fungible {
+		/// Check if the caller is the owner of the contract.
+		fn ensure_owner(&self) -> Result<(), Psp22Error> {
+			if self.owner != self.env().caller() {
+				return Err(Psp22Error::Custom(String::from("Not the owner")));
+			}
+			Ok(())
 		}
 
+		/// Transfer the ownership of the contract to another account.
+		///
+		/// # Parameters
+		/// - `owner` - New owner account.
 		#[ink(message)]
-		pub fn token_decimals(&self, token: TokenId) -> Result<u8> {
-			api::token_decimals(token)
+		pub fn transfer_ownership(&mut self, owner: AccountId) -> Result<(), Psp22Error> {
```

Comment:
Transfer is so similar to the language use for token transfer

```suggestion
		pub fn change_ownership(&mut self, owner: AccountId) -> Result<(), Psp22Error> {
```